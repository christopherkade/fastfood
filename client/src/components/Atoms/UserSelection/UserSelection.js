import React from "react"
import styled from "styled-components"

const Wrapper = styled.div`
  width: 100%;
`

const SelectionBox = styled.input`
  width: 50px;
  height: 50px;
  background-color: white;
  border: 1px solid white;
  font-size: 1.5rem;
  text-align: center;
  margin: 0.1rem;

  &:focus {
    outline: transparent;
  }

  &:hover {
    cursor: default;
  }
`

const UserSelection = ({ selections }) => {
  return (
    <Wrapper>
      {selections.map((selection, index) => <SelectionBox key={index} value={selection.emoji} readOnly />)}
    </Wrapper>
  )
}

export default UserSelection
