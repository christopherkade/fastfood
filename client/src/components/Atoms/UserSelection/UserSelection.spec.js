import React from 'react'
import { render, cleanup } from 'react-testing-library'

import UserSelection from "./UserSelection"

afterEach(cleanup);

describe("<UserSelection />", () => {
  let selections = null

  beforeEach(() => {
    selections = [
      {
        emoji: "🍕",
        name: "Pizza"
      }
    ]
  })

  describe("Snapshots", () => {
    it("should match snapshot", () => {
      const { container } = render(<UserSelection selections={selections} />);
      expect(container).toMatchSnapshot();
    });
  })
})
