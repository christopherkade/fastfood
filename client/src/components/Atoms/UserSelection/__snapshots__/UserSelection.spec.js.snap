// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserSelection /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa UoSJW"
  >
    <input
      class="sc-bwzfXH jmWvNq"
      readonly=""
      value="🍕"
    />
  </div>
</div>
`;
